src/values/AbstractObjectValue.js

```diff
@@ -257,7 +263,13 @@ export default class AbstractObjectValue extends AbstractValue {
     if (P instanceof StringValue) P = P.value;
 
     if (this.values.isTop()) {
-      AbstractValue.reportIntrospectionError(this, P);
+      let error = new CompilerDiagnostic(
+        "property access on unknown object",
+        this.$Realm.currentLocation,
+        "PP0031",
```

Comment:
Every error message should have a unique error code and wiki page discussing it. While these two errors are closely related, they arise in different situations can do with distinct discussion in the Wiki.

Reply:
I was mainly concerned about the prototype one. The others in this PR are basically the same error.

Reply:
Your wiki pages would be more helpful if you can add a code sample that illustrates what can go wrong.